apps/dashboard/src/app/(dashboard)/dashboard/connect/ecosystem/create/components/client/create-ecosystem-form.client.tsx

```diff
@@ -0,0 +1,192 @@
+"use client";
+
+import { ConfirmationDialog } from "@/components/ui/ConfirmationDialog";
+import { Button } from "@/components/ui/button";
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { ImageUpload } from "@/components/ui/image-upload";
+import { Input } from "@/components/ui/input";
+import { RadioGroup, RadioGroupItemButton } from "@/components/ui/radio-group";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { Loader2 } from "lucide-react";
+import Link from "next/link";
+import { useRouter } from "next/navigation";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+import { toast } from "sonner";
+import invariant from "tiny-invariant";
+import { z } from "zod";
+import { useCreateEcosystem } from "../../hooks/use-create-ecosystem";
+
+const formSchema = z.object({
+  name: z.string().min(1, {
+    message: "Ecosystem name is required",
+  }),
+  logo: z.instanceof(File, {
+    message: "Logo is required",
+  }),
+  permission: z.union([z.literal("PARTNER_WHITELIST"), z.literal("ANYONE")]),
+});
+
+const nameToSlug = (name = "") => `${name.toLowerCase().replaceAll(" ", "-")}`;
```

Comment:
this has to happen on the backend not on the frontend (since these have to be unique in DB)

Reply:
still relevant